Having mavenLocal first is bad practice.

codeconsole opened the pull request.

codeconsole committed:
Having mavenLocal first is bad practice.

matrei commented:
I think it's bad practice to commit `mavenLocal()` period.

But, in this case it is needed, and it is needed to be first (or at least before, the groovy snapshot repo), because the `groovy-joint-workflow` uses `mavenLocal()` to pull the groovy libs that was built in CI.

Preferably `mavenLocal()` should be removed and added by the `groovy-joint-workflow` workflow when it is run.

codeconsole commented:
> I think it's bad practice to commit `mavenLocal()` period.

@matrei I agree and understand why it's there. The issue I was having is unexpected old versions of things were taking preference because of it being first. Rather than purge my mavenLocal or re-build everything trying to find the root cause, I moved it after and used 6.0.1-SNAPSHOT in my demo ;)

> Preferably mavenLocal() should be removed and added by the groovy-joint-workflow workflow when it is run.

That's a good idea

matrei commented:
#13681
